mule-module-apikit/src/main/java/org/mule/module/apikit/transform/ApikitResponseTransformer.java

```diff
@@ -169,23 +166,72 @@ public String apply(MediaType mediaType)
      *
      * @return null if it is not
      */
-    private String acceptedContentType(String msgMimeType, String msgContentType, Collection<String> conjunctionTypes)
+    private String acceptedContentType(String msgMimeType, String encoding, String msgContentType, Collection<String> conjunctionTypes)
     {
         for (String acceptedMediaType : conjunctionTypes)
         {
-            if(msgMimeType != null && msgMimeType.contains(acceptedMediaType))
+            if (areCompatibleTypes(msgMimeType, acceptedMediaType))
             {
-                return msgMimeType;
+                return appendEncoding(encoding, acceptedMediaType);
             }
         }
         for (String acceptedMediaType : conjunctionTypes)
         {
-            if(msgContentType != null && msgContentType.contains(acceptedMediaType))
+            if (areCompatibleTypes(msgContentType, acceptedMediaType))
             {
-                return msgContentType;
+                final String contentTypeEncoding = extractEncoding(msgContentType);
+                return appendEncoding(contentTypeEncoding, acceptedMediaType);
             }
         }
         return null;
     }
 
+    private String appendEncoding(String encoding, String mimeType) {
+        return encoding != null && mimeType != null && !mimeType.contains(CHARSET_PARAMETER) ? mimeType + CHARSET_PARAMETER + encoding : mimeType;
+    }
+
+    private String extractEncoding(String msgContentType) {
+        final int charsetKeyIndex = msgContentType.indexOf(CHARSET_PARAMETER);
+
+        if (charsetKeyIndex == -1) return null;
+
+        final int endCharsetDefinition = msgContentType.indexOf(";", charsetKeyIndex + 1);
+
+        return endCharsetDefinition != -1 ? msgContentType.substring(charsetKeyIndex + CHARSET_PARAMETER.length(), endCharsetDefinition) : msgContentType.substring(charsetKeyIndex + CHARSET_PARAMETER.length());
+    }
+
+
+    private boolean areCompatibleTypes(String mimeType1, String mimeType2)
```

Comment:
*/* case is not contemplated

Reply:
I guess you mean "*/*" case, I will add this case as well